src/index.ts

```diff
@@ -121,26 +146,45 @@ async function run() {
   }
 }
 
-run().catch(async e => {
-  core.setFailed(e)
+async function post() {
+  await deleteKeychain()
+}
 
-  if (e instanceof SyntaxError && e.stack) {
-    core.error(e.stack)
+async function main() {
+  const isPost = Boolean(core.getState('isPost'))
+  if (isPost) {
+    return await post()
+  } else {
+    core.saveState('isPost', true)
```

Comment:
This confuses me can you explain?

Reply:
I'll add comments to clarify.

In order to use the same entry point for `main` and `post`, we `saveState` in `main` to be read by `post`. I followed the [logic](https://github.com/actions/checkout/blob/b2e6b7ed13bcde9d37c9e3e6967cd3ecfd2807ad/src/state-helper.ts) (though not the style) of `actions/checkout`.

I wanted to keep the same entry point in order to avoid duplicating error handling logic.